<a href="https://colab.research.google.com/github/V9502996099/ML/blob/main/lab1/%D0%97%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Задание по практике 1

##Загрузка датасета
В файле classification datasets расположены ссылки на датасеты. Выберите один в соответствии со своим вариантом, скачайте и прочитайте файл

**Не забудьте загрузить нужные библиотеки**

Например: pip3 install pandas

In [ ]:
import pandas as pd

In [ ]:
df = pd.read_csv("/content/Airplane_Crashes_and_Fatalities_Since_1908.csv")
df

Date   Time                                           Location  \
0     09/17/1908  17:18                                Fort Myer, Virginia   
1     07/12/1912  06:30                            AtlantiCity, New Jersey   
2     08/06/1913    NaN                 Victoria, British Columbia, Canada   
3     09/09/1913  18:30                                 Over the North Sea   
4     10/17/1913  10:30                         Near Johannisthal, Germany   
...          ...    ...                                                ...   
5263  05/20/2009  06:30                             Near Madiun, Indonesia   
5264  05/26/2009    NaN                  Near Isiro, DemocratiRepubliCongo   
5265  06/01/2009  00:15  AtlantiOcean, 570 miles northeast of Natal, Br...   
5266  06/07/2009  08:30       Near Port Hope Simpson, Newfoundland, Canada   
5267  06/08/2009    NaN                  State of Arunachal Pradesh, India   

                             Operator Flight #  \
0                Military - U.S. Army      NaN   
1                Military - U.S. Navy      NaN   
2                             Private        -   
3              Military - German Navy      NaN   
4              Military - German Navy      NaN   
...                               ...      ...   
5263  Military - Indonesian Air Force      NaN   
5264                      Service Air      NaN   
5265                       Air France      447   
5266                       Strait Air      NaN   
5267      Military - Indian Air Force      NaN   

                                          Route  \
0                                 Demonstration   
1                                   Test flight   
2                                           NaN   
3                                           NaN   
4                                           NaN   
...                                         ...   
5263                           Jakarta - Maduin   
5264                               Goma - Isiro   
5265                     Rio de Janeiro - Paris   
5266  Lourdes de BlanSablon - Port Hope Simpson   
5267                         Mechuka for Jorhat   

                                  Type Registration cn/In  Aboard  Fatalities  \
0                     Wright Flyer III          NaN     1     2.0         1.0   
1                            Dirigible          NaN   NaN     5.0         5.0   
2                     Curtiss seaplane          NaN   NaN     1.0         1.0   
3               Zeppelin L-1 (airship)          NaN   NaN    20.0        14.0   
4               Zeppelin L-2 (airship)          NaN   NaN    30.0        30.0   
...                                ...          ...   ...     ...         ...   
5263           Lockheed C-130 Hercules       A-1325  1982   112.0        98.0   
5264                     Antonov An-26       9Q-CSA  5005     4.0         4.0   
5265                   Airbus A330-203       F-GZCP   660   228.0       228.0   
5266  Britten-Norman BN-2A-27 Islander      C-FJJR    424     1.0         1.0   
5267                     Antonov An-32          NaN   NaN    13.0        13.0   

      Ground                                            Summary  
0        0.0  During a demonstration flight, a U.S. Army fly...  
1        0.0  First U.S. dirigible Akron exploded just offsh...  
2        0.0  The first fatal airplane accident in Canada oc...  
3        0.0  The airship flew into a thunderstorm and encou...  
4        0.0  Hydrogen gas which was being vented was sucked...  
...      ...                                                ...  
5263     2.0  While on approach, the military transport cras...  
5264     NaN  The cargo plane crashed while on approach to I...  
5265     0.0  The Airbus went missing over the AtlantiOcean ...  
5266     0.0  The air ambulance crashed into hills while att...  
5267     0.0  The military transport went missing while en r...  

[5268 rows x 13 columns]

##Первое знакомство с датасетом
В следующих ячейках выведите:


1.   Колонки датасета
2.   Первые 5 значений
3.   Последние 5 значений
4.   Статистику по числовым колонкам



In [ ]:
#Колонки
df.head()

Date   Time                            Location  \
0  09/17/1908  17:18                 Fort Myer, Virginia   
1  07/12/1912  06:30             AtlantiCity, New Jersey   
2  08/06/1913    NaN  Victoria, British Columbia, Canada   
3  09/09/1913  18:30                  Over the North Sea   
4  10/17/1913  10:30          Near Johannisthal, Germany   

                 Operator Flight #          Route                    Type  \
0    Military - U.S. Army      NaN  Demonstration        Wright Flyer III   
1    Military - U.S. Navy      NaN    Test flight               Dirigible   
2                 Private        -            NaN        Curtiss seaplane   
3  Military - German Navy      NaN            NaN  Zeppelin L-1 (airship)   
4  Military - German Navy      NaN            NaN  Zeppelin L-2 (airship)   

  Registration cn/In  Aboard  Fatalities  Ground  \
0          NaN     1     2.0         1.0     0.0   
1          NaN   NaN     5.0         5.0     0.0   
2          NaN   NaN     1.0         1.0     0.0   
3          NaN   NaN    20.0        14.0     0.0   
4          NaN   NaN    30.0        30.0     0.0   

                                             Summary  
0  During a demonstration flight, a U.S. Army fly...  
1  First U.S. dirigible Akron exploded just offsh...  
2  The first fatal airplane accident in Canada oc...  
3  The airship flew into a thunderstorm and encou...  
4  Hydrogen gas which was being vented was sucked...

In [ ]:
#Первые 5 значений
df[:3]
df.head(3)


Date   Time                            Location  \
0  09/17/1908  17:18                 Fort Myer, Virginia   
1  07/12/1912  06:30             AtlantiCity, New Jersey   
2  08/06/1913    NaN  Victoria, British Columbia, Canada   

               Operator Flight #          Route              Type  \
0  Military - U.S. Army      NaN  Demonstration  Wright Flyer III   
1  Military - U.S. Navy      NaN    Test flight         Dirigible   
2               Private        -            NaN  Curtiss seaplane   

  Registration cn/In  Aboard  Fatalities  Ground  \
0          NaN     1     2.0         1.0     0.0   
1          NaN   NaN     5.0         5.0     0.0   
2          NaN   NaN     1.0         1.0     0.0   

                                             Summary  
0  During a demonstration flight, a U.S. Army fly...  
1  First U.S. dirigible Akron exploded just offsh...  
2  The first fatal airplane accident in Canada oc...

In [ ]:
#Последние 5 значений
df[len(df)-5:len(df)]
df.tail()

Date   Time                                           Location  \
5263  05/20/2009  06:30                             Near Madiun, Indonesia   
5264  05/26/2009    NaN                  Near Isiro, DemocratiRepubliCongo   
5265  06/01/2009  00:15  AtlantiOcean, 570 miles northeast of Natal, Br...   
5266  06/07/2009  08:30       Near Port Hope Simpson, Newfoundland, Canada   
5267  06/08/2009    NaN                  State of Arunachal Pradesh, India   

                             Operator Flight #  \
5263  Military - Indonesian Air Force      NaN   
5264                      Service Air      NaN   
5265                       Air France      447   
5266                       Strait Air      NaN   
5267      Military - Indian Air Force      NaN   

                                          Route  \
5263                           Jakarta - Maduin   
5264                               Goma - Isiro   
5265                     Rio de Janeiro - Paris   
5266  Lourdes de BlanSablon - Port Hope Simpson   
5267                         Mechuka for Jorhat   

                                  Type Registration cn/In  Aboard  Fatalities  \
5263           Lockheed C-130 Hercules       A-1325  1982   112.0        98.0   
5264                     Antonov An-26       9Q-CSA  5005     4.0         4.0   
5265                   Airbus A330-203       F-GZCP   660   228.0       228.0   
5266  Britten-Norman BN-2A-27 Islander      C-FJJR    424     1.0         1.0   
5267                     Antonov An-32          NaN   NaN    13.0        13.0   

      Ground                                            Summary  
5263     2.0  While on approach, the military transport cras...  
5264     NaN  The cargo plane crashed while on approach to I...  
5265     0.0  The Airbus went missing over the AtlantiOcean ...  
5266     0.0  The air ambulance crashed into hills while att...  
5267     0.0  The military transport went missing while en r...

In [ ]:
#Статистику
df.describe()

Aboard   Fatalities       Ground
count  5246.000000  5256.000000  5246.000000
mean     27.554518    20.068303     1.608845
std      43.076711    33.199952    53.987827
min       0.000000     0.000000     0.000000
25%       5.000000     3.000000     0.000000
50%      13.000000     9.000000     0.000000
75%      30.000000    23.000000     0.000000
max     644.000000   583.000000  2750.000000

**Ответьте на следующие вопросы:**


1.   Какие средние значения в каждой числовой колонке вашего датасета?
2.   Минимальные и максимальные значения в каждой числовой колонке вашего датасета





1.   **Ответ 1.**

       Aboard 	  Fatalities   Ground

mean 	27.554518 	20.068303 	1.608845
2.   **Ответ 2.**

       Aboard 	Fatalities 	Ground

min 	0.000000 	0.000000 	0.000000

max 	644.000000 	583.000000 	2750.000000

In [ ]:
df.mean()

<ipython-input-49-c61f0c8f89b5>:1: FutureWarning: The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.mean()


Aboard        27.554518
Fatalities    20.068303
Ground         1.608845
dtype: float64

In [ ]:
df.min()

<ipython-input-50-c3612c624a3f>:1: FutureWarning: The default value of numeric_only in DataFrame.min is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.min()


Date          01/01/1966
Aboard               0.0
Fatalities           0.0
Ground               0.0
dtype: object

In [ ]:
df.max()

<ipython-input-51-4c1ddf8920ff>:1: FutureWarning: The default value of numeric_only in DataFrame.max is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.max()


Date          12/31/1997
Aboard             644.0
Fatalities         583.0
Ground            2750.0
dtype: object

##Типизация
Присвойте соответствующим колонкам один из типов (category, float, int)

In [ ]:
"""
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   Date          5268 non-null   object
 1   Time          3049 non-null   object
 2   Location      5248 non-null   object
 3   Operator      5250 non-null   object
 4   Flight #      1069 non-null   object
 5   Route         3562 non-null   object
 6   Type          5241 non-null   object
 7   Registration  4933 non-null   object
 8   cn/In         4040 non-null   object
 9   Aboard        5246 non-null   float64
 10  Fatalities    5256 non-null   float64
 11  Ground        5246 non-null   float64
 12  Summary       4878 non-null   object

Место для вашего кода
"""
df.Location = df.Location.astype("category")
#Вывести типы колонок тут
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5268 entries, 0 to 5267
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype   
---  ------        --------------  -----   
 0   Date          5268 non-null   object  
 1   Time          3049 non-null   object  
 2   Location      5248 non-null   category
 3   Operator      5250 non-null   object  
 4   Flight #      1069 non-null   object  
 5   Route         3562 non-null   object  
 6   Type          5241 non-null   object  
 7   Registration  4933 non-null   object  
 8   cn/In         4040 non-null   object  
 9   Aboard        5246 non-null   float64 
 10  Fatalities    5256 non-null   float64 
 11  Ground        5246 non-null   float64 
 12  Summary       4878 non-null   object  
dtypes: category(1), float64(3), object(9)
memory usage: 666.9+ KB


Какие признаки вы можете выделить в своем датасете (категориальные, числовые и др., **минимум 3 разных**, помните что можно также выделить признаки более точно, например числовые - порядковые, количественные и т.д.)?

Location - категориальные
Fatalities - численные


##Фильтрация
Офильтруйте 3 колонки вашего датасета и выведите результат

In [ ]:
#Фильтр колонки 1
df[df["Aboard"] < 2]

Date   Time                                      Location  \
2     08/06/1913    NaN            Victoria, British Columbia, Canada   
21    12/16/1918    NaN                         Elizabeth, New Jersey   
22    05/25/1919    NaN                               Cleveland, Ohio   
23    07/19/1919    NaN                         Dix Run, Pennsylvania   
24    10/02/1919    NaN                            Newcastle, England   
...          ...    ...                                           ...   
5185  01/14/2008  05:08                              Off Lihu, Hawaii   
5195  04/09/2008  23:27                           Bundeena, Australia   
5202  05/23/2008  01:25                             Billings, Montana   
5210  06/18/2008  10:03                        Hyannis, Massachusetts   
5266  06/07/2009  08:30  Near Port Hope Simpson, Newfoundland, Canada   

                           Operator Flight #  \
2                           Private        -   
21           US Aerial Mail Service      NaN   
22           US Aerial Mail Service      NaN   
23           US Aerial Mail Service      NaN   
24    Aircraft Transport and Travel      NaN   
...                             ...      ...   
5185                Apline Aviation      NaN   
5195             Avtex Air Services      NaN   
5202               Alpine Aviation      5008   
5210                Wiggins Airways      NaN   
5266                     Strait Air      NaN   

                                          Route  \
2                                           NaN   
21                                          NaN   
22                                          NaN   
23                                          NaN   
24                                          NaN   
...                                         ...   
5185                               Lihu - Kauai   
5195                          Sydney - Brisbane   
5202                     Billings - Great Falls   
5210                        Hyannis - Nantucket   
5266  Lourdes de BlanSablon - Port Hope Simpson   

                                          Type Registration   cn/In  Aboard  \
2                             Curtiss seaplane          NaN     NaN     1.0   
21                           De Havilland DH-4           97     NaN     1.0   
22                           De Havilland DH-4           61     NaN     1.0   
23                           De Havilland DH-4           82     NaN     1.0   
24                           De Havilland DH-4          NaN     NaN     1.0   
...                                        ...          ...     ...     ...   
5185                        Beechcraft 1900C-1      N410UB    UC-70     1.0   
5195         Swearingen SA227AC Metroliner III       VH-OZA  AC-600     1.0   
5202                          Beechcraft 1900C       N195GA    U-65     1.0   
5210  de Havilland Canada DHC-6 Twin Otter 100       N656WA      47     1.0   
5266          Britten-Norman BN-2A-27 Islander      C-FJJR      424     1.0   

      Fatalities  Ground                                            Summary  
2            1.0     0.0  The first fatal airplane accident in Canada oc...  
21           1.0     0.0                                                NaN  
22           1.0     0.0  Caught fire in midair. The pilot leaped from t...  
23           1.0     0.0                                                NaN  
24           1.0     0.0                                                NaN  
...          ...     ...                                                ...  
5185         1.0     0.0  The cargo plane was carrying mail when it cras...  
5195         1.0     0.0  The pilot of the mail plane reported some mino...  
5202         1.0     0.0  After taking off and  being told to turn left,...  
5210         1.0     0.0  The cargo plane took off, reached about 200 fe...  
5266         1.0     0.0  The air ambulance crashed into hills while att...  

[166 rows x 13 columns]

In [ ]:
#Фильтр колонки 2
df[df["Ground"] >= 1.0]

Date   Time                   Location  \
55    08/24/1921  17:40      River Humber, England   
58    01/14/1922    NaN              Paris, France   
296   03/25/1933  20:07        Hayward, California   
356   05/18/1935  00:45        Near Moscow, Russia   
433   05/06/1937  19:25      Lakehurst, New Jersey   
...          ...    ...                        ...   
5206  05/30/2008  09:40      Tegucigalpa, Honduras   
5217  07/07/2008  03:50      Near Bogota, Colombia   
5236  11/04/2008  18:40        Mexico City, Mexico   
5248  02/12/2009  22:17  Clarence Center, New York   
5263  05/20/2009  06:30     Near Madiun, Indonesia   

                               Operator Flight #  \
55       Military - Royal Airship Works      NaN   
58               Handley Page Transport      NaN   
296                     Varney Airlines      NaN   
356                  Gor'ky Eskadril'ya      NaN   
433          Deutsche Zeppelin Reederei   LZ 129   
...                                 ...      ...   
5206        TACA International Airlines      390   
5217                Centurian Air Cargo      NaN   
5236                 Mexican Government      NaN   
5248  Continental Connection/Colgan Air     3407   
5263    Military - Indonesian Air Force      NaN   

                                                  Route  \
55                                                  NaN   
58                                                  NaN   
296                             Burbank - San Francisco   
356                                       Demonstration   
433                          Frankfurt - Lakehurst N.J.   
...                                                 ...   
5206  San Salvador, El Salvador - Tegucigalpa, Hondu...   
5217                       Bogota, Colombia - Miami, FL   
5236                      San Luis Potosi - Mexico City   
5248                         Newark, N.J. - Buffalo, NY   
5263                                   Jakarta - Maduin   

                                    Type Registration      cn/In  Aboard  \
55    Royal Airship Works ZR-2 (airship)         R-38       R-38    46.0   
58                     Handley Page O/10       G-EATN      HP-43     5.0   
296                     Lockheed Orion 9      NC12226        184     3.0   
356                 Tupolev ANT-20 / I-5     CCCP-I20        NaN    50.0   
433                      Zeppelin LZ-129     D-LZ-129    D-LZ129    97.0   
...                                  ...          ...        ...     ...   
5206                    Airbus A-320-233       EI-TAF       1374   136.0   
5217                 Boeing B-747-209BSF       N714CK  22446/519     8.0   
5236                          Learjet 45       XC-VMC     45-028     8.0   
5248           Bombardier DHC-8-402 Q400       N200WQ       4200    49.0   
5263             Lockheed C-130 Hercules       A-1325       1982   112.0   

      Fatalities  Ground                                            Summary  
55          46.0     1.0  Crashed due to structural failure followed by ...  
58           5.0     5.0                       Crashed at Le Bourget field.  
296          3.0    11.0  The aircraft crashed into tops of two houses a...  
356         50.0     2.0  Midair collision with an I-5 fighter at 2,300 ...  
433         35.0     1.0  The airship Hindenburg caught fire and burned ...  
...          ...     ...                                                ...  
5206         3.0     2.0  After making a missed approach and during the ...  
5217         0.0     3.0  The cargo plane carrying a load of flowers had...  
5236         8.0     8.0  Mexican Interior Minister Juan Camilo Mourino,...  
5248        49.0     1.0  The commuter plane crashed while attemptiong t...  
5263        98.0     2.0  While on approach, the military transport cras...  

[219 rows x 13 columns]

In [ ]:
#Фильтр колонки 3
df[df["Fatalities"] == 0.0]

Date   Time                                      Location  \
108   10/21/1926  13:15                               English Channel   
387   03/05/1936    NaN                            Near Tengya, China   
889   10/08/1947  07:55                           Near El Paso, Texas   
897   11/11/1947    NaN                            Gallup, New Mexico   
1265  05/25/1953  15:30                        Amsterdam, Netherlands   
1359  01/19/1955  16:25                              Des Moines, Iowa   
1440  09/09/1956    NaN                        Bartlesville, Oklahoma   
1443  10/16/1956  06:15                          Over the PacifiOcean   
1610  06/30/1959  10:35         Ishikawa City, Ryukyu Island, Okinawa   
1837  02/28/1963  21:29                         Boston, Massachusetts   
1868  08/21/1963  12:15                       Near Leningrad , Russia   
1885  12/17/1963  23:41                       Los Angeles, California   
1927  07/16/1964  18:29                            Richmond, Virginia   
1982  06/28/1965  13:10                     San Francisco, California   
1983  07/01/1965  05:29                         Kansas City, Missouri   
2066  08/17/1966    NaN                   Near Da Nang, South Vietnam   
2247  08/28/1968  07:25                             Lopez, Washington   
2266  11/22/1968  01:24                     San Francisco, California   
2359  12/22/1969  10:30                    Near San Diego, California   
2486  07/30/1971  14:29                     San Francisco, California   
2590  09/24/1972    NaN                        Sacramento, California   
2835  08/07/1975  16:11                              Denver, Colorado   
3182  11/15/1979  12:00                        Near Chicago, Illinois   
3307  09/22/1981  12:00                         Near Babaeski, Turkey   
3341  06/24/1982  20:44                   Mount Galunggung, Indonesia   
3366  07/23/1982  02:30                           Castaic, California   
3417  05/05/1983  09:15          En route Miami, FL - Nassau, Bahamas   
3428  07/23/1983  19:30                       Gimli, Manitoba, Canada   
3470  02/28/1984  16:16               JFK Airport, New York, New York   
3541  02/19/1985  10:15        300 nm NW of San Francisco, California   
3549  04/16/1985  11:37                        Las Cruces, New Mexico   
3611  03/27/1986    NaN              Bangui, Central African Republic   
3767  05/24/1988  12:55                        New Orleans, Louisiana   
3927  03/01/1990    NaN                            Near Meerut, India   
3950  06/10/1990  07:33                          Oxfordshire, England   
4068  12/27/1991  08:51                             Stockholm, Sweden   
4117  07/30/1992  17:41               JFK Airport, New York, New York   
4171  03/31/1993  12:25                             Anchorage, Alaska   
4199  08/18/1993  16:56  Leeward Point Airfield, Guantanamo Bay, Cuba   
4231  03/02/1994  17:59                            New York, New York   
4242  03/23/1994  14:30             Near Fayetteville, North Carolina   
4273  08/10/1994  11:22                            Cheju, South Korea   
4339  06/08/1995  19:08                              Atlanta, Georgia   
4543  02/03/1998  15:15                  Near Cavalese, Trento, Italy   
4553  03/22/1998  19:42                          Bacolod, Philippines   
4594  10/21/1998  13:51                              Yerevan, Armenia   
4701  03/05/2000  18:11                          Burbank, Calilfornia   
4748  10/06/2000  16:55                               Reynosa, Mexico   
4797  08/24/2001  18:46                              Terceira, Azores   
4875  07/27/2002  12:45                                 Lviv, Ukraine   
5057  08/02/2005  16:04                               Toronto, Canada   
5074  12/08/2005  19:15                             Chicago, Illinois   
5134  01/25/2007  11:15                                   Pau, France   
5178  11/08/2007  08:00                               Khartoum, Sudan 

##Группировка
Сгруппируйте датасет по какому-либо категориальному признаку и выведите среднее значение численных колонок (см. метод mean в pandas)

In [ ]:
#Группировка
df.groupby(["Location"]).count()

Date  Time  Operator  Flight #  \
Location                                                                   
1,200 miles off Dakar, AtlantiOcean          1     0         1         0   
100 miles SW of Kuujjuaq, Quebec, Canada     1     1         1         1   
110 miles SW of Sochi, Russia                1     1         1         1   
200 miles NE of Derby, Australia             1     1         1         0   
25 nm off Agrigento, Italy                   1     1         1         1   
...                                        ...   ...       ...       ...   
Zheng Zou AB, China                          1     0         1         0   
Zifta, Egypt                                 1     1         1         0   
Zurich, Switzerland                          6     3         6         2   
Zwingle, Iowa                                1     1         1         1   
near Orillia, Ontario, Canada                1     1         1         0   

                                          Route  Type  Registration  cn/In  \
Location                                                                     
1,200 miles off Dakar, AtlantiOcean           1     1             1      1   
100 miles SW of Kuujjuaq, Quebec, Canada      0     1             1      1   
110 miles SW of Sochi, Russia                 1     1             1      1   
200 miles NE of Derby, Australia              0     1             1      1   
25 nm off Agrigento, Italy                    0     1             1      0   
...                                         ...   ...           ...    ...   
Zheng Zou AB, China                           1     1             1      0   
Zifta, Egypt                                  1     1             1      0   
Zurich, Switzerland                           4     6             6      6   
Zwingle, Iowa                                 1     1             1      1   
near Orillia, Ontario, Canada                 1     1             1      1   

                                          Aboard  Fatalities  Ground  Summary  
Location                                                                       
1,200 miles off Dakar, AtlantiOcean            1           1       1        1  
100 miles SW of Kuujjuaq, Quebec, Canada       1           1       1        1  
110 miles SW of Sochi, Russia                  1           1       1        1  
200 miles NE of Derby, Australia               1           1       1        1  
25 nm off Agrigento, Italy                     1           1       1        0  
...                                          ...         ...     ...      ...  
Zheng Zou AB, China                            1           1       1        1  
Zifta, Egypt                                   1           1       1        1  
Zurich, Switzerland                            6           6       6        6  
Zwingle, Iowa                                  1           1       1        1  
near Orillia, Ontario, Canada                  1           1       1        1  

[4303 rows x 12 columns]

##Задание для гениев *
Агрегируйте числовой признак по категориальному признаку с использованием метода **без использования** библиотеки pandas, и вычислите медиану и среднее значение для каждой категории.

In [ ]:
#Агрегация